Compile errors with fmt 10.1.0

Hi,

both HEAD and tag v1.2.0 do not build with fmt version 10.1.0. I stumbled over this due to Arch bumping its package, but this also occurs if I bump the version of fmt in `etc/repos.json`. It builds with 10.0.0, so this is not a major issue right now, but you will probably need to address this before updating fmt the next time.

The relevant error message is added below. 

```
In file included from /tmp/tmpgtv8ewdx/src/src/buildtool/main/main.cpp:24:
In file included from /tmp/tmpgtv8ewdx/src/src/buildtool/build_engine/base_maps/entity_name.hpp:24:
In file included from /tmp/tmpgtv8ewdx/src/src/buildtool/build_engine/base_maps/entity_name_data.hpp:25:
In file included from /tmp/tmpgtv8ewdx/src/src/buildtool/build_engine/expression/expression_ptr.hpp:24:
In file included from /tmp/tmpgtv8ewdx/src/src/buildtool/build_engine/expression/function_map.hpp:21:
In file included from /tmp/tmpgtv8ewdx/src/src/buildtool/build_engine/expression/linked_map.hpp:26:
/tmp/tmp3ltxf4ic/dep_includes/fmt/core.h:1577:63: error: implicit instantiation of undefined template
      'fmt::detail::type_is_unformattable_for<const nlohmann::basic_json<>, char>'
    type_is_unformattable_for<T, typename Context::char_type> _;
                                                              ^
/tmp/tmp3ltxf4ic/dep_includes/fmt/core.h:1810:23: note: in instantiation of function template specialization
      'fmt::detail::make_arg<true, fmt::basic_format_context<fmt::appender, char>, const nlohmann::basic_json<>, 0>' requested here
        data_{detail::make_arg<is_packed, Context>(args)...} {
                      ^
/tmp/tmp3ltxf4ic/dep_includes/fmt/core.h:1828:10: note: in instantiation of function template specialization
      'fmt::format_arg_store<fmt::basic_format_context<fmt::appender, char>, nlohmann::basic_json<>,
      nlohmann::basic_json<>>::format_arg_store<const nlohmann::basic_json<>, const nlohmann::basic_json<>>' requested here
  return {args...};
         ^
/tmp/tmp3ltxf4ic/src/src/buildtool/logging/logger.hpp:130:48: note: in instantiation of function template specialization
      'fmt::make_format_args<fmt::basic_format_context<fmt::appender, char>, const nlohmann::basic_json<>, const nlohmann::basic_json<>>'
      requested here
            auto fmsg = fmt::vformat(msg, fmt::make_format_args(args...));
                                               ^
/tmp/tmp3ltxf4ic/src/src/buildtool/logging/logger.hpp:93:13: note: in instantiation of function template specialization
      'Logger::FormatAndForward<const nlohmann::basic_json<> &, const nlohmann::basic_json<> &>' requested here
            FormatAndForward(nullptr,
            ^
/tmp/tmp3ltxf4ic/src/src/buildtool/main/main.cpp:475:17: note: in instantiation of function template specialization
      'Logger::Log<const nlohmann::basic_json<> &, const nlohmann::basic_json<> &>' requested here
        Logger::Log(LogLevel::Error,
                ^
/tmp/tmp3ltxf4ic/dep_includes/fmt/core.h:1555:45: note: template is declared here
template <typename T, typename Char> struct type_is_unformattable_for;
                                            ^
/tmp/tmp3ltxf4ic/dep_includes/fmt/core.h:1580:3: error: static assertion failed due to requirement 'formattable': Cannot
      format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
  static_assert(
  ^
/tmp/tmp3ltxf4ic/dep_includes/fmt/core.h:1577:63: error: implicit instantiation of undefined template
      'fmt::detail::type_is_unformattable_for<nlohmann::basic_json<>, char>'
    type_is_unformattable_for<T, typename Context::char_type> _;
                                                              ^
/tmp/tmp3ltxf4ic/dep_includes/fmt/core.h:1810:23: note: in instantiation of function template specialization
      'fmt::detail::make_arg<true, fmt::basic_format_context<fmt::appender, char>, nlohmann::basic_json<>, 0>' requested here
        data_{detail::make_arg<is_packed, Context>(args)...} {
                      ^
/tmp/tmp3ltxf4ic/dep_includes/fmt/core.h:1828:10: note: in instantiation of function template specialization
      'fmt::format_arg_store<fmt::basic_format_context<fmt::appender, char>, nlohmann::basic_json<>, std::basic_string<char>,
      std::basic_string<char>>::format_arg_store<nlohmann::basic_json<>, const std::basic_string<char>, const std::basic_string<char>>'
      requested here
  return {args...};
         ^
/tmp/tmp3ltxf4ic/src/src/buildtool/logging/logger.hpp:130:48: note: in instantiation of function template specialization
      'fmt::make_format_args<fmt::basic_format_context<fmt::appender, char>, nlohmann::basic_json<>, const std::basic_string<char>, const
      std::basic_string<char>>' requested here
            auto fmsg = fmt::vformat(msg, fmt::make_format_args(args...));
                                               ^
/tmp/tmp3ltxf4ic/src/src/buildtool/logging/logger.hpp:93:13: note: in instantiation of function template specialization
      'Logger::FormatAndForward<nlohmann::basic_json<> &, const std::basic_string<char> &, const std::basic_string<char> &>' requested here
            FormatAndForward(nullptr,
            ^
/tmp/tmp3ltxf4ic/src/src/buildtool/main/main.cpp:654:29: note: in instantiation of function template specialization
      'Logger::Log<nlohmann::basic_json<> &, const std::basic_string<char> &, const std::basic_string<char> &>' requested here
                    Logger::Log(LogLevel::Error,
                            ^
/tmp/tmp3ltxf4ic/dep_includes/fmt/core.h:1555:45: note: template is declared here
template <typename T, typename Char> struct type_is_unformattable_for;
                                            ^
/tmp/tmp3ltxf4ic/dep_includes/fmt/core.h:1580:3: error: static assertion failed due to requirement 'formattable': Cannot
      format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt
  static_assert(
  ^
4 errors generated.
```

--- a/src/utils/cpp/TARGETS
+++ b/src/utils/cpp/TARGETS
@@ -16,7 +16,12 @@
   { "type": ["@", "rules", "CC", "library"]
   , "name": ["json"]
   , "hdrs": ["json.hpp"]
-  , "deps": ["gsl", ["@", "json", "", "json"], ["@", "gsl", "", "gsl"]]
+  , "deps":
+    [ "gsl"
+    , ["@", "json", "", "json"]
+    , ["@", "gsl", "", "gsl"]
+    , ["@", "fmt", "", "fmt"]
+    ]
   , "stage": ["src", "utils", "cpp"]
   }
 , "concepts":

--- a/src/utils/cpp/json.hpp
+++ b/src/utils/cpp/json.hpp
@@ -21,6 +21,7 @@
 #include <string>
 #include <unordered_map>
 
+#include "fmt/ostream.h"
 #include "gsl/gsl"
 #include "nlohmann/json.hpp"
 #include "src/utils/cpp/gsl.hpp"
@@ -211,5 +212,11 @@ namespace detail {
     }
     return old_abbrev;
 }
+
+// Use nlohmann::basic_json::operator<<() for formatting via libfmt.
+// This explicit template specialization seems to be required starting with
+// libfmt 10.x.
+template <>
+struct fmt::formatter<nlohmann::basic_json<>> : ostream_formatter {};
 
 #endif  // INCLUDED_SRC_UTILS_CPP_JSON_HPP

